controller/controller-worker/src/main/java/com/pinterest/rocksplicator/controller/tasks/RecurTask.java

```diff
@@ -118,6 +118,9 @@ public void process(Context ctx) throws Exception {
     try {
       ctx = new Context(id, ctx.getCluster(), localAckTaskQueue, ctx.getWorker());
       task.process(ctx);
+      // in case the current body is changed, and we need to carry the state
+      // to next enqueued job
+      getParameter().getTask().body = task.getParameter().serialize();
```

Comment:
can we do getParameter().setTask(task) ?

Reply:
use setBody() instead because Task's other 2 fields are name and priority